runbooks/workflow/creating-maintaining-littlepay-data-syncs.md

```diff
@@ -0,0 +1,51 @@
+
+## Setting Up a New Littlepay Sync
+
+### Prerequisites
+
+In order to access source data from Littlepay, you will need a vendor-provided AWS access key for the agency you're setting up. Examples of access keys for existing Littlepay data syncs can be found in [Google Cloud Secret Manager](https://console.cloud.google.com/security/secret-manager?project=cal-itp-data-infra).
+
+You'll need to install the `aws` CLI locally, and [configure a profile](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-quickstart.html) for each Littlepay `merchant_id` (aka `account name` or `instance` in Littlepay).
+
+You can confirm you have access to a given access key by listing the keys for the specific instance using the profile you configured:
+`aws iam list-access-keys --user-name <username> --profile <profile>`. **Note that the username may not be exactly the same as the `merchant_id`; check the key JSON or error output for the username.** For example, `An error occurred (AccessDenied) when calling the ListAccessKeys operation: User: arn:aws:iam::840817857296:user/system/sbmtd-default is not authorized to perform: iam:ListAccessKeys on resource: user sbmtd because no identity-based policy allows the iam:ListAccessKeys action` indicates that the username is `sbmtd-default`.
+
+### First-Time Setup Steps
+
+When you receive an access key from Littlepay for an agency that is being newly synced, you need to add that access key to Google Cloud Secret Manager, named with the format `LITTLEPAY_AWS_IAM_<MERCHANT_ID>_ACCESS_KEY` like the existing keys. Secret names are uppercase, and we have removed hyphens (or convert to underscore) in the `merchant_id` historically.
+
+Next, [create a service account in the cal-itp-data-infra project](https://console.cloud.google.com/iam-admin/serviceaccounts/create?walkthrough_id=iam--create-service-account&project=cal-itp-data-infra#step_index=1) specifically for this agency's Littlepay sync, using the naming convention `<MERCHANT_ID>-payments-user`. There is a pre-configured role titled Agency Payments Service User, which will give the new service account the correct permissions when assigned to that account.
+
+After that, you should create a simple YAML file in [the sync-littlepay DAG folder in the data-infra repository](https://github.com/cal-itp/data-infra/tree/8516a1a4ab2ecfe6ef33e3fbc4224bcedbd06e98/airflow/dags/sync_littlepay), structurally similar to the others that are there, utilizing the instance name and access key name for the agency you're setting up. Similarly, you should create a YAML file corresponding to the new agency in [the parse-littlepay DAG folder](https://github.com/cal-itp/data-infra/tree/8516a1a4ab2ecfe6ef33e3fbc4224bcedbd06e98/airflow/dags/parse_littlepay). Then, add a mapping between the agency's instance identifier and its Cal-ITP internal GTFS dataset identifier in [this seed file](https://github.com/cal-itp/data-infra/blob/main/warehouse/seeds/payments_gtfs_datasets.csv).
```

Comment:
Rather than framing this as "create a YAML file", could it be reframed as "create the sync and parse Airflow DAG tasks" to emphasize the function/purpose of the YAML?

Reply:
Resolved in [0049f24](https://github.com/cal-itp/data-infra/pull/3019/commits/0049f24032f29d2a243d2db04fe472dcd4067ae7)